src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/UpdateVCFSequenceDictionaryIntegrationTest.java

```diff
@@ -83,14 +88,45 @@ private void testBadUpdateSequenceDictionary(
             final File inputVariantsFile,
             final File inputSourceFile,
             final File inputReferenceFile,
+            final String masterSequenceDictionary,
             final boolean replace) {
-        updateSequenceDictionary(inputVariantsFile, inputSourceFile, inputReferenceFile, replace);
+        updateSequenceDictionary(inputVariantsFile, inputSourceFile, inputReferenceFile, masterSequenceDictionary, replace);
+    }
+
+    @Test
+    public void testUseMasterDictionary() {
+        final SAMSequenceDictionary actualSequenceDictionary = updateSequenceDictionary(
+                new File(testDir, "variantsNoDict.vcf"),
+                null,
+                null,
+                new File(testDir, "exampleFASTA.dict").getAbsolutePath(),
+                false);
+        final SAMSequenceDictionary expectedSequenceDictionary = SAMSequenceDictionaryExtractor.extractDictionary(
+                Paths.get(new File(testDir, "exampleFASTA.dict").getAbsolutePath()));
+
+        // verify only the sequence names and lengths, since other attributes such as MD/UR will have been updated
```

Comment:
@cmnbroad Is it worth verifying that the other attributes are different?

Reply:
I don't think it is. In real life usage, whether or not these attributes these attributes change depends on the state of the original inputs. Since this test uses a starting vcf that has no sequence dictionary at all, the MD5 and UR attributes get added on, but we just need to validate that the correct SD got propagated.